<a href="https://colab.research.google.com/github/Kashaf537/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Kashaf537/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane: Content Refresh Prioritization

I selected the **Content Refresh Prioritization** lane because websites often contain hundreds or thousands of pages, making it difficult for content teams to decide which pages should be updated first. My goal is to use the available page performance metrics to identify pages that may require refreshing so editors can focus on the highest-priority content. This problem is suitable for machine learning because multiple factors such as impressions, CTR, rankings, and content age may jointly influence page performance.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

**Decision**

Which content pages should be refreshed first to maintain or improve search performance?

**Who acts on the output?**

SEO specialists or content editors will use the recommendations.

**Action**

The recommended pages will be reviewed, updated, or optimized before their performance declines further.

**Cost of a wrong recommendation**

If a page is incorrectly identified as needing a refresh, editors may spend time updating content that does not require changes. If a declining page is missed, the website may continue losing traffic and ranking opportunities.

**Why can machine learning help?**

Page performance depends on several metrics rather than a single factor. Machine learning can combine multiple signals to prioritize pages more effectively than a simple manual rule.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
!git clone https://github.com/Kashaf537/flyrank-ml-internship.git

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 123, done.
remote: Counting objects: 100% (123/123), done.
remote: Compressing objects: 100% (75/75), done.
remote: Total 123 (delta 39), reused 105 (delta 32), pack-reused 0 (from 0)
Receiving objects: 100% (123/123), 1.84 MiB | 14.10 MiB/s, done.
Resolving deltas: 100% (39/39), done.


In [9]:
!ls flyrank-ml-internship/data/raw

content_refresh_anonymized.csv


In [10]:
import pandas as pd

df = pd.read_csv(
    "flyrank-ml-internship/data/raw/content_refresh_anonymized.csv"
)
print("Total Pages:", len(df))
print("Average CTR:", round(df["ctr"].mean(),3))
print("Average Content Age (days):", round(df["content_age_days"].mean(),1))

print("\nTrend Direction Counts")
print(df["trend_direction"].value_counts())

Total Pages: 30000
Average CTR: 0.511
Average Content Age (days): 256.2

Trend Direction Counts
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


## Observations

-The dataset contains 30,000 content pages, providing a large sample for analysis.

-The average CTR is 0.511%, and the average content age is approximately 256 days, indicating that the dataset contains pages with different levels of search performance and age.

-The trend_direction distribution shows that 16,262 pages are declining, which is the largest category.

-This suggests that prioritizing declining pages could be a useful decision-support problem for content teams.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This work provides decision support by identifying patterns observed in the available dataset. The findings are directional rather than causal, meaning they can help prioritize pages that may require attention based on historical performance data.

This project does not prove that updating a page will improve its Google ranking, nor does it predict Google's search algorithm. The results should be used to support human decision-making rather than replace it, and any recommendations should be validated before taking action.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.